eng/Build.ps1

```diff
@@ -44,7 +44,7 @@ param (
     [switch]$procdump,
     [switch]$deployExtensions,
     [switch]$prepareMachine,
-    [switch]$useGlobalNuGetCache = $true,
+    [switch]$useGlobalNuGetCache = $false,
```

Comment:
Why is this needed? Won't that slow down CI steps each time?

Reply:
This is in fact a correct change.

As it stands at the moment, there is no way to not use the global cache, because:
the command: ````build```` and the command: ````build -useGlobalNuGetCache```` both $useGlobalNuGetCache set to true

With this change the command: ````build```` sets $useGlobalNuGetCache to $False and the command:
````build -useGlobalNuGetCache```` sets $useGlobalNuGetCache to $True

The reason it's important is that in order to ensure that our build is not subject to supply chain attacks, I.e.  Where packages are replaced by packages from different '**malicious feeds**' we control the contents of the packages.config file amongst other things.  We also would like to ensure that when we build, we download the specific packages from exactly the specified feed and not some either feed via the the global nuget.

This does mean that on a local build after this change the command: ````build```` will cause the creation of this directory ````blah!!! ... \fsharp\.packages```` in which all of the downloaded nuget packages will be stored.

It is annoying though that the by-default behavior on developer machines will now change.
One way to achieve that is by 'deprecating the useGlobalNuGetCache' switch and adding a dontUseGlobalNuGetCache switch.

This switch would override the default: ````useGlobalNuGetCache```` setting it to false.  This would ensure that developers got the behavior they have now.  Whilst the ci build would be able to set the ````dontUseGlobalNuGetCache```` to get the good supply chain safe approach.

I think I will propose a PR to change this.

Thoughts
Kevin

Reply:
@T-Gro : This is resolved by:  https://github.com/dotnet/fsharp/pull/14029/commits/037310c64e67cfc66cad4e45dc31cc908babd98b